Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -66,84 +63,112 @@ + (instancetype)orderedSetWithObjects:(id)firstObj {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (instancetype)orderedSetWithObjects:(id _Nonnull const[])objects count:(NSUInteger)cnt {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (instancetype)orderedSetWithObjects:(id _Nonnull const[])objects count:(NSUInteger)count {
+    return [[[NSOrderedSet alloc] initWithObjects:objects count:count] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithOrderedSet:(NSOrderedSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [NSOrderedSet orderedSetWithOrderedSet:set range:NSMakeRange(0, [set count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithOrderedSet:(NSOrderedSet*)set range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] initWithOrderedSet:set range:range copyItems:flag] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithSet:(NSSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] initWithSet:set] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithSet:(NSSet*)set copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] initWithSet:set copyItems:flag] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (range.length > [array count]) {
```

Comment:
This check only works as intended when range.location == 0. Change to NSMaxRange(range)